### PR TITLE
fix: return no data if calculation does not exist for requested period

### PR DESCRIPTION
**Pull request by MadsDue**
# Description

If we can't find any calculation for the request periode and calculation type we now return "No Data" instead of throwing missing calculation exception. 

However, if we finds calculation for part of the requested period, we throw exception with missing calculation. 

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002` or `sandbox_002`)
- [ ] Documentation has been updated
- [ ] The integration [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) has been updated
- [ ] C4 diagrams have been updated and Team Outlaws has been informed about relevant changes

### CODE REVIEW

File: source/dotnet/wholesale-api/Edi/AggregatedTimeSeriesRequestHandler.cs [hunk 1]
```diff
@@ -73,35 +73,49 @@ public async Task ProcessAsync(ServiceBusReceivedMessage receivedMessage, string
...
-        var parameters = await CreateAggregatedTimeSeriesQueryParametersWithoutCalculationTypeAsync(request).ConfigureAwait(false);
+        var error = new List<ValidationError> { _noDataAvailable };
+        if (await EnergySupplierOrBalanceResponsibleHaveAggregatedTimeSeriesForAnotherGridAreasAsync(aggregatedTimeSeriesRequest, aggregatedTimeSeriesRequestMessage).ConfigureAwait(false))
+        {
+            error = [_noDataForRequestedGridArea];
+        }
 
-        return await _aggregatedTimeSeriesQueries.GetAsync(
-            parameters).ToListAsync(cancellationToken).ConfigureAwait(false);
+        _logger.LogInformation("No data available for AggregatedTimeSeriesRequest message with reference id {reference_id}", referenceId);
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/AggregatedTimeSeriesRequestHandler.cs:117]: Why were we loggin this to begin with? I struggle to find a scenario where I would need it.

File: source/dotnet/wholesale-api/Edi/Calculations/LatestCalculationsForPeriod.cs [hunk 2]
```diff
@@ -58,9 +58,21 @@ public LatestCalculationsForPeriod(DateTimeZone dateTimeZone)
                 return latestCalculationsForPeriod.OrderBy(x => x.Period.Start).ToList();
         }
 
-        return latestCalculationsForPeriod.Count == 0
-            ? (IReadOnlyCollection<CalculationForPeriod>)latestCalculationsForPeriod.OrderBy(x => x.Period.Start).ToList()
-            : throw new MissingCalculationException($"No calculation found for dates: {string.Join(", ", remainingDaysInPeriod)}");
+        if (DaysInPeriodWithNoCalculations(remainingDaysInPeriod, latestCalculationsForPeriod))
+        {
+            throw new MissingCalculationException($"No calculation found for dates: {string.Join(", ", remainingDaysInPeriod)}");
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/Calculations/LatestCalculationsForPeriod.cs:63]: If this happens - would we not want to expose more information about the exeception? Perhaps the requested calculation_type and the grid area for a start?

File: source/dotnet/wholesale-api/Edi/Calculations/LatestCalculationsForPeriod.cs [hunk 3]
```diff
@@ -58,9 +58,21 @@ public LatestCalculationsForPeriod(DateTimeZone dateTimeZone)
...
+        return latestCalculationsForPeriod.OrderBy(x => x.Period.Start)
+            .ToList();
+    }
+
+    /// <summary>
+    /// Ensure that all days in the period have a calculation or none have a calculation.
+    /// </summary>
+    private static bool DaysInPeriodWithNoCalculations(List<Instant> remainingDaysInPeriod, List<CalculationForPeriod> latestCalculationsForPeriod)
+    {
+        return remainingDaysInPeriod.Any() && latestCalculationsForPeriod.Any();
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/Calculations/LatestCalculationsForPeriod.cs:75]: None of these need to compared? I would assume that this method would make sure that each remaning day is within a calculationperiod - or am I misunderstanding?

File: source/dotnet/wholesale-api/Edi/AggregatedTimeSeriesRequestHandler.cs (hunk 1 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Edi/AggregatedTimeSeriesRequestHandler.cs:117]: Maybe a leftover from when the connection between the two systems were established.

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**